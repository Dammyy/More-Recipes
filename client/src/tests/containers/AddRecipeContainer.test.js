import React from 'react';
import { shallow } from 'enzyme';
import { AddRecipeContainer }
  from '../../containers/AddRecipeContainer';

describe('AddRecipeContainer', () => {
  describe('Snapshot', () => {
    test('component should fully render', () => {
      const props = {
        recipesActions: { addRecipe: jest.fn(), },
        filestackActions: { uploadImage: jest.fn(), },
        image: ''
      };
      const component = shallow(<AddRecipeContainer {...props} />);
      expect(component).toMatchSnapshot();
    });
  });
  describe('Instance methods', () => {
    describe('AddRecipeContainer instance methods', () => {
      test('', () => {
        const props = {
          recipesActions: { addRecipe: jest.fn(), },
          filestackActions: { uploadImage: jest.fn(), },
          image: '',
          handleSubmit: jest.fn(),
        };
        const component = shallow(<AddRecipeContainer {...props} />);
        component.instance().uploadImage();
        component.instance().handleSubmit({ preventDefault: () => {} });
        expect(props.recipesActions.addRecipe).toBeCalled();
      });
    });
  });
});
