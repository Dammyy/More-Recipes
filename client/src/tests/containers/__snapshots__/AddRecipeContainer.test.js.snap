// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRecipeContainer Snapshot component should fully render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddRecipeContainer
    filestackActions={
      Object {
        "uploadImage": [MockFunction],
      }
    }
    image=""
    recipesActions={
      Object {
        "addRecipe": [MockFunction],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "handleSubmit": [Function],
      "image": "",
      "uploadImage": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "handleSubmit": [Function],
        "image": "",
        "uploadImage": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
